app/models/add_spiked_in_control_task.rb

```diff
@@ -5,27 +5,56 @@
 # been added to the lane in question.
 # @see Tasks::AddSpikedInControlHandler for behaviour included in the {WorkflowsController}
 class AddSpikedInControlTask < Task
+  # Holds information about a PhiX buffer for a given request in order to render the view.
+  # - If indirect, will inform the user that PhiX has already been added, but will let them
+  # override this with a deliberate click.
+  # - To partial path is used by rails if you do `render object`, this lets us keep the logic
+  #   outside of the views.
+  SpikedBufferRecord =
+    Struct.new(:barcode, :indirect, :request) do
+      def to_partial_path
+        indirect ? 'indirect_phi_x_fields' : 'direct_phi_x_fields'
+      end
+
+      def label
+        request ? "Request #{request.position} : #{request.asset.display_name} PhiX Barcode" : 'PhiX Barcode'
+      end
+    end
+
   def partial
     'add_spiked_in_control'
   end
 
-  def do_task(controller, params)
-    controller.do_add_spiked_in_control_task(self, params)
+  def can_process?(batch, from_previous: false)
+    batch.released? ? [true, 'Edit'] : [true, nil]
   end
 
-  def add_control(batch, control_asset, request_id_set)
-    return false unless batch && control_asset
+  def do_task(workflows_controller, params)
+    Tasks::AddSpikedInControlHandler::Handler.new(controller: workflows_controller, params: params, task: self).perform
+  end
 
-    batch.requests.each do |request|
-      next unless request_id_set.include? request.id
+  def fields_for(requests)
```

Comment:
To be honest, most of this added code would better be placed on a presenter, but doing that properly will involve updating all the tasks.